core/network/Cargo.toml

```diff
@@ -46,6 +46,7 @@ erased-serde = "0.3.9"
 void = "1.0.2"
 zeroize = "0.10.1"
 babe-primitives = { package = "substrate-consensus-babe-primitives", path = "../consensus/babe/primitives" }
+state-machine = { package = "substrate-state-machine", path = "../../core/state-machine" }
```

Comment:
new deps could probably be avoided.

Reply:
We can re-export from client, but I don't see the reason. `substrate-state-machine` is an implicit dependency anyway, why not make it explicit?

Reply:
semantically 'network' depending on 'state-machine' feels a bit odd. I would rather reexport from client.